File: code/cpp/ch14/practice/warmup_04_is_prefix.cpp
/*
 * Warmup 4: Is Array Prefix of Another
 * Chapter 14: Prefix Sums — The Running Total Trick
 *
 * PROBLEM: Return true if arr1 is a prefix of arr2.
 *
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */

#include <algorithm>
#include <climits>
#include <iostream>
#include <vector>
using namespace std;

bool solve(vector<int> arr1, vector<int> arr2) {
    // TODO: Replace this with your solution
    return false;
}

// -- Do not change anything below this line --------------------------
int main() {
    int n1, n2;
    cin >> n1;
    vector<int> arr1(n1);
    for (int i = 0; i < n1; i++) cin >> arr1[i];
    cin >> n2;
    vector<int> arr2(n2);
    for (int i = 0; i < n2; i++) cin >> arr2[i];
    cout << (solve(arr1, arr2) ? "true" : "false") << endl;
    return 0;
}

File: code/java/ch14/practice/Warmup04IsPrefix.java
package ch14.practice;

import java.util.*;

/**
 * Warmup 4: Is Array Prefix of Another
 * Chapter 14: Prefix Sums — The Running Total Trick
 *
 * PROBLEM: Return true if arr1 is a prefix of arr2.
 *
 * EXAMPLES:
 *   solve([1,2,3], [1,2,3,4,5]) -> true
 *   solve([1,2,4], [1,2,3,4,5]) -> false
 *   solve([], [1,2,3])           -> true
 *
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */
public class Warmup04IsPrefix {
    public static boolean solve(int[] arr1, int[] arr2) {
        // TODO: Replace this with your solution
        return false;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String line1 = sc.nextLine().trim();
        String line2 = sc.nextLine().trim();
        int[] arr1 = line1.isEmpty() ? new int[]{} : Arrays.stream(line1.split(" ")).mapToInt(Integer::parseInt).toArray();
        int[] arr2 = line2.isEmpty() ? new int[]{} : Arrays.stream(line2.split(" ")).mapToInt(Integer::parseInt).toArray();
        System.out.println(solve(arr1, arr2));
        sc.close();
    }
}
